Create a way to get provider metadata when using the MultiSamlStrategy

mlunoe opened the pull request:
This adds a function to pass in a request along with the other `generateServiceProviderMetadata` arguments to retrieve provider metadata when using the MultiSamlStrategy. If there is no request, we cannot call the `_getSamlOptions`-function to retrieve all the necessary options to call the `generateServiceProviderMetadata`-function with.

mlunoe commented:
@markstos It would be wonderful to get this change in and get an updated version with it, so I can install it into our application. Is there anything else you need from me on this change?

markstos commented:
@mlunoe Please add tests and documentation for your commit and squash the result into a single commit.

In the meantime, you can use the `package.json` syntax to load a dependency straight from your Github repo instead of NPM.

mlunoe commented:
That tests and description was a requirement would have been valuable information up front, so I did not have to wait :/

Ok, I have added tests and a description and squashed commits

markstos commented:
Thanks. I've made a note to a do a final review and release.

mlunoe commented:
I made an update to embraced the callback structure that the `getSamlOptions` function requires and added tests to cover this change.

markstos commented:
@weikaolun @stavros-wb Thanks for the peer-reviews.

@mlunoe This is being merged now. A release is expected within a week, as some other PRs are under review as well.

cjbarth commented:
@markstos I see that a release was expected within a week that would contain this fix, but that release is pending some other code being reviewed. I can help with reviews if that helps get this code out. I could use this for my project too.

cjbarth commented:
I found a bug with this feature. I'll make a PR soon to fix it.

markstos commented:
Thanks @cjbarth I'll wait for that.

markstos commented:
@cjbarth There are some failing tests on the master branch need review and fixing for a release to happen. Also, taking a look at https://github.com/bergie/passport-saml/pull/341 would be helpful.

cjbarth commented:
I was noticing the failing tests. Do we know when they started failing?

cjbarth commented:
@markstos , I've created PR #367 to fix the broken tests. Once that lands, if @Archinowsk wants to merge `master` into his branch for #341, that should fix his issues with broken tests. If not, I can have a look at that too. I would love to get this all wrapped up early next week if possible :)

Archinowsk commented:
> Once that lands, if @Archinowsk wants to merge master into his branch for #341, that should fix his issues with broken tests. If not, I can have a look at that too.

Please do if you are available :).

markstos commented:
v1.1 was released today.
